api/internal/owner/serializers.py

```diff
@@ -143,7 +143,11 @@ def validate_value(self, value):
         return value
 
     def validate(self, plan):
-        owner = self.context["view"].owner
+        current_org = self.context["view"].owner
+        if current_org.account:
+            raise serializers.ValidationError(
+                detail="You cannot update your plan manually, for help or changes to plan, connect with sales@codecov.io"
+            )
```

Comment:
heheh, this is definitely a django serializers thing. there are different flows to serialize incoming and outgoing objects. the object only goes through `validate` when you try to update it, so when the object is incoming to the database. It doesn't go through here on the way out to be displayed.

Reply:
Oooff, this is definitely not intuitive.

Would we also want to pull some values in this "plan" from the `accounts` object instead? i.e. plan type, billing rate?

Reply:
heheheh, it is :)

Reply:
so for Orgs, we need to think about the GET experience and the PATCH experience, (no POSTs to this endpoint)

the GET experience (what we return to the User to see), they see the `pretty_plan` from the `Account` if the Org has an `Account`, if they don't have an account they see the `pretty_plan` on the Org.

The PATCH experience (user wants to update plan):
- without Account: can update the plan on the Org according to all the layered rules and validation we have already 😅 
- with Account: firstly, they shouldn't be able to see options to update the plan since they are sales led, so this should never happen, but just in case they manage to send this in, reject changes with this error message.

Reply:
this logic is twisty-turny because things like the `pretty_plan` replacement are happening in `shared` and plan updating comes through `gazebo` (+1 for monorepo!!), lmk if you wanna do a call to trace the pathways 👍 👍 👍